include/dataformats/TriggerRecordHeader.hpp

```diff
@@ -11,48 +11,236 @@
+  run_number_t get_run_number() const { return header_()->run_number; }
+  /**
+   * @brief Set the run number for this TriggerRecordHeader
+   * @param run_number Run number to set
+  */
+  void set_run_number(run_number_t run_number) { header_()->run_number = run_number; }
+  /**
+   * @brief Get the error_bits header field as a bitset
+   * @return bitset containing error_bits header field
+  */
+  std::bitset<32> get_error_bits() const { return header_()->error_bits; }
+  /**
+   * @brief Overwrite error bits using the given bitset
+   * @param bits Bitset of error bits to set
+  */
+  void set_error_bits(std::bitset<32> bits) { header_()->error_bits = bits.to_ulong(); }
+  /**
+   * @brief Get the value of the given error bit
+   * @param bit Bit to get
+   * @return Value of bit (true/false)
+  */
+  bool get_error_bit(size_t bit) const { return get_error_bits()[bit]; }
+  /**
+   * @brief Set the given error bit to the given value
+   * @param bit Bit to set
+   * @param value Value to set (true/false)
+  */
+  void set_error_bit(size_t bit, bool value)
+  {
+    auto bits = get_error_bits();
+    bits[bit] = value;
+    set_error_bits(bits);
+  }
+
+  /**
+   * @brief Get the trigger_type field from the data struct
+   * @return The trigger_type field from the TriggerRecordHeaderData struct
+  */
+  trigger_type_t get_trigger_type() const { return header_()->trigger_type; }
+  /**
+   * @brief Set the trigger_type header field to the given value
+   * @param trigger_type Value of trigger_type to set
+  */
+  void set_trigger_type(trigger_type_t trigger_type) { header_()->trigger_type = trigger_type; }
+
+  /**
+   * @brief Get the total size of the TriggerRecordHeader
+   * @return The size of the TriggerRecordHeader, including header and all component requests
+   */
+  size_t get_total_size_bytes() const
+  {
+    return header_()->num_requested_components * sizeof(ComponentRequest) + sizeof(TriggerRecordHeaderData);
+  }
+  /**
+   * @brief Get the location of the flat data array for I/O
+   * @return Pointer to the TriggerRecordHeader data array
+  */
+  void* get_storage_location() const { return data_arr_; }
+
+  /**
+   * @brief Access ComponentRequest and copy result
+   * @param idx Index to access
+   * @return Copy of ComponentRequest at index
+   * @throws ERS error if idx is outside of allowable range
+  */
+  ComponentRequest at(size_t idx) const
+  {
+    if (idx >= header_()->num_requested_components) {
+      // Throw ERS exception
+    }
+    // Increment header pointer by one to skip header
+    return *(reinterpret_cast<ComponentRequest*>(header_() + 1) + idx); // NOLINT
+  }
+
+  /**
+   * @brief Operator[] to access ComponentRequests by index
+   * @param idx Index to access
+   * @return ComponentRequest reference
+   * @throws ERS error if idx is outside of allowable range
+  */
+  ComponentRequest& operator[](size_t idx)
+  {
+    if (idx >= header_()->num_requested_components) {
+      // Throw ERS exception
```

Comment:
This is a TODO, complicated by the fact that this code will be shared with offline